Entities/Common/Building/BlockPlacement.as

```diff
@@ -44,7 +44,7 @@ void PlaceBlock(CBlob@ this, u8 index, Vec2f cursorPos)
 		server_TakeRequirements(inv, bc.reqs);
 		getMap().server_SetTile(cursorPos, bc.tile);
 
-		u32 delay = this.get_u32("build delay");
+		u32 delay = (getRules().hasTag("faster building") ? this.get_u32("warmup build delay") : this.get_u32("build delay"));
```

Comment:
This statement is duplicated in a few places, this should be refactored into a function in a common script (`getCurrentBuildSpeed()` or something)

Reply:
fixed